bundles/core/org.eclipse.smarthome.core.audio/src/main/java/org/eclipse/smarthome/core/audio/internal/AudioManagerImpl.java

```diff
@@ -90,51 +91,89 @@ public void play(AudioStream audioStream) {
 
     @Override
     public void play(AudioStream audioStream, String sinkId) {
-        if (audioStream != null) {
-            AudioSink sink = getSink(sinkId);
+        play(audioStream, sinkId, null);
+    }
+
+    public void play(AudioStream audioStream, String sinkId, PercentType volume) {
+        if (audioStream == null) {
+            throw new NullPointerException("Audio stream cannot be played as it is null.");
+        }
 
-            if (sink != null) {
+        AudioSink sink = getSink(sinkId);
+        if (sink != null) {
+            // get current volume
+            PercentType oldVolume = getVolume(sinkId);
+            // set notification sound volume
+            if (volume != null) {
+                setVolume(volume, sinkId);
+            }
+            if (sink.getSupportedStreams().stream().anyMatch(clazz -> clazz.isInstance(audioStream))) {
                 try {
                     sink.process(audioStream);
                 } catch (UnsupportedAudioFormatException | UnsupportedAudioStreamException e) {
-                    logger.error("Error playing '{}': {}", audioStream.toString(), e.getMessage());
+                    logger.warn("Error playing '{}': {}", audioStream, e.getMessage(), e);
                 }
             } else {
-                logger.warn("Failed playing audio stream '{}' as no audio sink was found.", audioStream.toString());
+                logger.warn("Failed playing audio stream '{}' as audio doesn't support it.", audioStream);
             }
+            // restore volume
+            if (oldVolume != null) {
+                setVolume(oldVolume, sinkId);
+            }
+        } else {
+            logger.warn("Failed playing audio stream '{}' as no audio sink was found.", audioStream);
         }
     }
 
     @Override
     public void playFile(String fileName) throws AudioException {
-        playFile(fileName, null);
+        playFile(fileName, null, null);
+    }
+
+    @Override
+    public void playFile(String fileName, PercentType volume) throws AudioException {
+        playFile(fileName, null, volume);
     }
 
     @Override
     public void playFile(String fileName, String sink) throws AudioException {
+        playFile(fileName, sink, null);
+    }
+
+    @Override
+    public void playFile(String fileName, String sink, PercentType volume) throws AudioException {
+        if (fileName == null) {
```

Comment:
Just a suggestion: `Objects#requireNonNull(Object, String)`

Reply:
Hm, ... I think when sca tool failed because of throwing NPEs it's not by intention of the designer to use a workaround instead. I changed my mind and will use annotations for both APIs.